### DISCUSSION TITLE
Pedagogical Interactive Notebooks, GSoC 2023 Project

### DISCUSSION THREAD

**Discussion by elshorbagyx** [networkx/networkx#6464] [General]
Hello there,
The description of the "Pedagogical Interactive Notebooks" project suggests that the contributor might implement 2 to 3 notebooks for a medium-sized project and 4 to 5 for a large-sized project. However, the completed projects section shows that similar projects were done with only one notebook. So which is right ? Or more specifically, how many notebooks should I include in the proposal?

Thanks for having me .

**Comment by elshorbagyx** [comment]
@rossbar  @MridulS

**Comment by MridulS** [comment]
We expect you to propose the number of topics as suggested on the ideas page. But once you start working on the project you may get stuck with something, maybe there is something wrong with networkx codebase itself which requires to go first look and research into the networkx codebase. These projects are flexible and we discuss these things over the summer. You don't have to stick a 100% to the proposal (it's perfectly fine if you want to!).

It's important for us that you are working on something you enjoy and the project can evolve over the summer :)

**Reply by elshorbagyx** [reply]
Very helpful ! Thanks alot :)
